service/controller/resource/heartbeatrouting/receiver/receiver.go

```diff
@@ -14,24 +15,39 @@ import (
 	"github.com/giantswarm/prometheus-meta-operator/service/key"
 )
 
-func toReceiver(cluster metav1.Object, installation string, opsgenieKey string) (alertmanagerconfig.Receiver, error) {
+func toReceiver(cfg alertmanagerconfig.Config, cluster metav1.Object, installation string, opsgenieKey string) (alertmanagerconfig.Receiver, error) {
 	u, err := url.Parse(fmt.Sprintf("https://api.opsgenie.com/v2/heartbeats/%s/ping", key.HeartbeatName(cluster, installation)))
 	if err != nil {
 		return alertmanagerconfig.Receiver{}, microerror.Mask(err)
 	}
 
+	httpConfig := &promcommonconfig.HTTPClientConfig{}
+	if cfg.Global != nil && cfg.Global.HTTPConfig != nil {
+		httpConfigCp, err := cloneHttpConfig(cfg.Global.HTTPConfig)
+		if err != nil {
+			return alertmanagerconfig.Receiver{}, err
+		}
+		httpConfig = httpConfigCp
+		// httpConfig = cfg.Global.HTTPConfig
+	}
+	httpConfig.BasicAuth = &promcommonconfig.BasicAuth{
+		Password: promcommonconfig.Secret(opsgenieKey),
+	}
+
+	if cfg.Global != nil && cfg.Global.HTTPConfig != nil {
+		if reflect.DeepEqual(cfg.Global.HTTPConfig, httpConfig) {
```

Comment:
Aren't two struct holding identical data but at different address equal ?
This is something that should be test and not done at runtime, because I do not see how copy can fail at runtime.

Reply:
Ditto, I only used it to make sure that the cloning works (e.g. this fails if I uncomment the line `httpConfig = cfg.Global.HTTPConfig`, but works with `httpConfig = httpConfigCp`), we don't need this at runtime so I'll remove this.

Reply:
Done.